config.c

```diff
@@ -787,84 +761,34 @@ bool load_config(char configPath[PATH_MAX], struct config_params *p, bool colors
-    GetPrivateProfileString("color", "gradient_color_2", "not_set", p->gradient_colors[1], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_3", "not_set", p->gradient_colors[2], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_4", "not_set", p->gradient_colors[3], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_5", "not_set", p->gradient_colors[4], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_6", "not_set", p->gradient_colors[5], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_7", "not_set", p->gradient_colors[6], 9,
-                            configPath);
-    GetPrivateProfileString("color", "gradient_color_8", "not_set", p->gradient_colors[7], 9,
-                            configPath);
-
-    p->horizontal_gradient = GetPrivateProfileInt("color", "horizontal_gradient", 0, configPath);
-
-    GetPrivateProfileString("color", "horizontal_gradient_color_1", "not_set",
-                            p->horizontal_gradient_colors[0], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_2", "not_set",
-                            p->horizontal_gradient_colors[1], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_3", "not_set",
-                            p->horizontal_gradient_colors[2], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_4", "not_set",
-                            p->horizontal_gradient_colors[3], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_5", "not_set",
-                            p->horizontal_gradient_colors[4], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_6", "not_set",
-                            p->horizontal_gradient_colors[5], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_7", "not_set",
-                            p->horizontal_gradient_colors[6], 9, configPath);
-    GetPrivateProfileString("color", "horizontal_gradient_color_8", "not_set",
-                            p->horizontal_gradient_colors[7], 9, configPath);
-
     if (strcmp(p->theme, "none") != 0) {
         configPath = configFileBak;
     }
-    free(p->theme);
 #endif
-    p->gradient_count = 0;
-    for (int i = 0; i < 7; ++i) {
-        if (strcmp(p->gradient_colors[i], "not_set") != 0)
-            p->gradient_count++;
-        else
-            break;
-    }
-    p->horizontal_gradient_count = 0;
-    for (int i = 0; i < 7; ++i) {
-        if (strcmp(p->horizontal_gradient_colors[i], "not_set") != 0)
-            p->horizontal_gradient_count++;
-        else
-            break;
-    }
-
     free(themeFile);
+    bool resultColor = load_colors(configPath, p, error);
```

Comment:
here there can be a:
```
bool resultColor;
    if (strcmp(p->theme, "none") != 0) {
        resultColor = load_colors(themeFile, p, error);
    } else {
    resultColor = load_colors(configPath, p, error);
}
```
or something

Reply:
p->color = strdup(iniparser_getstring(ini, "color:foreground", "default"));
    p->bcolor = strdup(iniparser_getstring(ini, "color:background", "default"));
    
    should also be in the load_colors function